src/main/controllers/StillWorkingController.ts

```diff
@@ -61,9 +61,12 @@ export default class StillWorkingController {
   }
 
   public post = async (req: AppRequest, res: Response): Promise<void> => {
-    if (req.body.isStillWorking === StillWorking.WORKING && req.session.userCase.endDate !== undefined) {
+    if (req.body.isStillWorking !== StillWorking.NO_LONGER_WORKING && req.session.userCase.endDate !== undefined) {
       req.session.userCase.endDate = undefined;
     }
+    if (req.body.isStillWorking !== StillWorking.NOTICE && req.session.userCase.noticeEnds !== undefined) {
```

Comment:
`&& req.session.userCase.noticeEnds !== undefined` is this bit necessary here? If we're resetting noticeEnds back to undefined anyway then it doesn't matter what the value is during the evaluation of this if statement